ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -757,6 +767,72 @@ private AdminResponse handleCatchupStatusRequest(DataInputStream requestStream,
     return new CatchupStatusAdminResponse(isCaughtUp, adminResponse);
   }
 
+  /**
+   * Handles {@link com.github.ambry.protocol.AdminRequestOrResponseType#BlobStoreControl}.
+   * @param requestStream the serialized bytes of the request.
+   * @param adminRequest the {@link AdminRequest} received.
+   * @return the {@link AdminResponse} to the request.
+   * @throws IOException if there is any I/O error reading from the {@code requestStream}.
+   */
+  private AdminResponse handleBlobStoreControlRequest(DataInputStream requestStream, AdminRequest adminRequest)
+      throws IOException {
+    ServerErrorCode error;
+    BlobStoreControlAdminRequest blobStoreControlAdminRequest =
+        BlobStoreControlAdminRequest.readFrom(requestStream, adminRequest);
+    if (blobStoreControlAdminRequest.getNumReplicasCaughtUpPerPartition() > 0) {
+      PartitionId partitionId = blobStoreControlAdminRequest.getPartitionId();
+      if (partitionId != null) {
+        error = validateRequest(partitionId, RequestOrResponseType.AdminRequest);
+        if (error.equals(ServerErrorCode.No_Error)) {
+          if (blobStoreControlAdminRequest.shouldEnable()) {
+            // TODO: start BlobStore properly
```

Comment:
at current time, return an error at this point. We can fill in the code in another PR.

Reply:
Error code added.